superset-frontend/src/dashboard/components/nativeFilters/state.ts

```diff
@@ -18,34 +18,61 @@
  */
 import { useSelector } from 'react-redux';
 import { useCallback, useMemo } from 'react';
-import { createSelector } from '@reduxjs/toolkit';
-import { Filter, Divider, isFilterDivider } from '@superset-ui/core';
+import {
+  Filter,
+  FilterConfiguration,
+  Divider,
+  isFilterDivider,
+  NativeFilterType,
+  ChartCustomization,
+  ChartCustomizationDivider,
+  ChartCustomizationConfiguration,
+} from '@superset-ui/core';
 import { ActiveTabs, DashboardLayout, RootState } from '../../types';
 import { CHART_TYPE, TAB_TYPE } from '../../util/componentTypes';
 
-const defaultFilterConfiguration: Filter[] = [];
+export function useFilterConfiguration() {
+  return useSelector<RootState, FilterConfiguration>(state => {
+    const filters = state.nativeFilters?.filters || {};
+    return Object.values(filters).filter(
+      (
+        item: Filter | Divider | ChartCustomization | ChartCustomizationDivider,
+      ): item is Filter | Divider =>
+        item.type === NativeFilterType.NativeFilter ||
+        item.type === NativeFilterType.Divider,
+    );
+  });
+}
 
-const selectFilterConfiguration = createSelector(
-  (state: RootState) =>
-    state.dashboardInfo?.metadata?.native_filter_configuration,
-  (nativeFilterConfig): (Filter | Divider)[] => {
-    if (!nativeFilterConfig) {
-      return defaultFilterConfiguration;
-    }
-    return nativeFilterConfig.filter(
-      (filter: any) => filter.type !== 'CHART_CUSTOMIZATION',
+export function useChartCustomizationConfiguration() {
+  const allCustomizations = useSelector<any, ChartCustomizationConfiguration>(
+    state => state.dashboardInfo.metadata?.chart_customization_config || [],
+  );
+
+  const dashboardLayout = useDashboardLayout();
+
+  return useMemo(() => {
+    const dashboardChartIds = new Set(
+      Object.values(dashboardLayout)
```

Comment:
Let's pre-filter dashboardLayout as this calculation might be expensive on a large dashboard